### PR TITLE
support default namespace for list api

### PR DESCRIPTION
**Pull request by nianliuu**
## Problem

in pinecone, the default namespace is "", the check in list will throw exception if namespace is ""

## Solution

remove the empty namespace check, only check namespace is null

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan
./gradlew build, passed all tests

### CODE REVIEW

**Comment by nianliuu** [comment]
@russcam can anyone help have a look of this PR? thank you!

**Comment by rohanshah18** [comment]
@nianliuu thank you for creating the PR, I have ran the integration test. Once successful, I'll approve the PR.

**Comment by nianliuu** [comment]
The API key is required and must not be empty or null, looks like API-Key not set for IT

**Review by rohanshah18 — APPROVED** [review]
I was able to successfully run the integration tests with these changes, LGTM! Once again, thank you for creating the PR.

**Pull request merged**
